ui-manchette/src/types.ts

```diff
@@ -51,7 +51,19 @@ export type OperationalPointType = ArrayElement<PathProperties['operational_poin
 export type StyledOperationalPointType = OperationalPointType & {
   styles?: CSSProperties;
   display?: boolean;
-  onClick?: (opId: string, opRef: HTMLDivElement | null) => void;
+  onClick?: (opId: string) => void;
+};
+
+export type WaypointMenuData = {
+  activeOperationalPointId?: string;
```

Comment:
Same comment as https://github.com/OpenRailAssociation/osrd-ui/pull/668/files#r1816147595

Could we maybe use an index instead of a custom string to refer to a particular OP in the list?

Reply:
Was afraid that when a waypoint is hidden when zooming out or when we hide one when clicking on it, it will break something but you're right it should be ok.

Reply:
Ah right, that's a good thing to keep in mind indeed! Maybe at some point we should introduce a unique waypoint ID (which would be different if the same OP appears twice), similar to OSRD's `PathItemLocation.id`?

But TBH, why does osrd-ui even need to access the OP ID? Couldn't we always pass a guaranteed unique waypoint ID in there? I suppose all of these questions are related to https://github.com/OpenRailAssociation/osrd-ui/issues/645.